app/Controller/Component/RoleComponent.php

```diff
@@ -1346,7 +1346,20 @@ public function isCouPerson($coPersonId, $coId, $couId) {
    */
   
   public function isGroupManager($coPersonId, $coGroupId) {
-    if(!$coPersonId || !$coGroupId) {
+    if(!$coGroupId) {
+      return false;
+    }
+
+    if($coPersonId === false) {
+      // The API User can be CO admin only if the co_id is equal to the
+      // co_id we calculated from the request data. In order for this
+      // to happen the Group and the API User must belong to the same CO
+      $roles = $this->calculateCMRoles();
```

Comment:
`calculateCMRoles()` isn't really intended to be called recursively, it's more for Controllers to call. I'd be a little concerned about side effects since this is old code. It might make more sense to just directly implement the logic you need here, eg:

```
if($this->Session->check('Auth.User.api')) {
  if($authCoId == 1 || (($coId == $authCoId) && $privd)) {...}
}
```

Or maybe refactor that particular section from `calculateCMRoles` into its own function that both `calculateCMRoles` and `isGroupManager` can call.